lib/dry/types/hash/schema.rb

```diff
@@ -149,8 +149,6 @@ def resolve(hash)
           end
         end
       end
-
-      private_constant(*constants(false))
```

Comment:
Alrighty. I played with rubinius and found that you are able to do `private_constant(:Foo, :Bar)` but my trick to do `private_constant(*constants(false))` doesn't work (the latter privatizes all constants already defined in the current scope). 

Anyways yes nothing is truly private in ruby and even with private constants you can still do `const_get`. Ping me if you think there is no way to keep them private. I put work into these classes and wanted to clarify their interface too which is why I chimed in here.